### PR TITLE
Improve inspect for multi platform container images

### PR DESCRIPTION
**Pull request by munishchouhan**
This PR will add a platform option in inspect API to get the correct details about a multi-arch container

### CODE REVIEW

**Comment by munishchouhan** [comment]
inspect view tested in local:
1. arm46

Screenshot 2024-09-27 at 14 41 02

2. amd 64

Screenshot 2024-09-27 at 15 20 11

File: src/main/groovy/io/seqera/wave/controller/ViewController.groovy [hunk 1]
```diff
@@ -242,10 +242,10 @@ class ViewController {
 
     @View("inspect-view")
     @Get('/inspect')
-    HttpResponse<Map<String,Object>> viewInspect(@QueryValue String image) {
+    HttpResponse<Map<String,Object>> viewInspect(@QueryValue String image, @Nullable @QueryValue String platform) {
         final binding = new HashMap(10)
         try {
-            final spec = inspectService.containerSpec(image, null)
+            final spec = inspectService.containerSpec(image, null, platform)
```
Comment (pditommaso) [inline src/main/groovy/io/seqera/wave/controller/ViewController.groovy:248]: Let's change the signature to  `image, platform, null`

File: src/main/groovy/io/seqera/wave/service/inspect/ContainerInspectService.groovy [hunk 2]
```diff
@@ -73,5 +73,5 @@ interface ContainerInspectService {
      * @return
      *      The {@link ContainerSpec} object modelling the container image inspect metadata
      */
-    ContainerSpec containerSpec(String containerImage, PlatformId identity)
+    ContainerSpec containerSpec(String containerImage, PlatformId identity, String platform)
```
Comment (pditommaso) [inline src/main/groovy/io/seqera/wave/service/inspect/ContainerInspectService.groovy:76]: Let's use `String containerImage, String platform, PlatformId identity` whenever possible

Comment (munishchouhan) [reply src/main/groovy/io/seqera/wave/service/inspect/ContainerInspectService.groovy]: done

Comment (pditommaso) [reply src/main/groovy/io/seqera/wave/service/inspect/ContainerInspectService.groovy]: Let's do also [here](https://github.com/seqeralabs/wave/pull/660/files#diff-16868f80a923c2d26d795773f1c8fc23cd1af435c9a469cc27bcee7f08186eb1R99) for consistency

Comment (munishchouhan) [reply src/main/groovy/io/seqera/wave/service/inspect/ContainerInspectService.groovy]: done

**Comment by pditommaso** [comment]
Well done

**Pull request merged**